feat(argocd): update chart to v5.14.1

hbollon opened the pull request:
With this PR I updated the ArgoCD's chart to the latest version since it haven't be done in a while and because it's needed to be able to use the [argocd-cmp-helmfile](https://github.com/camptocamp/docker-argocd-cmp-helmfile) plugin (and for the use of sidecar container for plugins in general).
I also changed the `HELM_VALUES` env variable to `ARGOCD_ENV_HELM_VALUES` to fix [this breaking change](https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/2.3-2.4/#update-plugins-to-use-newly-prefixed-environment-variables) in the latest argocd image version (a new `camptocamp/argocd` version was released FYI).

Be aware that few major/breaking changes were introduced in the lot of versions of ArgoCD's chart/image that we have behind so it may have some incompatibilities with some of the deployed platforms.

Additionally, do you know why do we do `helm dependency build || true` for kustomized-helm plugin? 
https://github.com/camptocamp/devops-stack/blob/e67c521bf2d77086b508b2c31a0e4889730024ee/modules/values.tmpl.yaml#L105
It seems to be a little non-sense and due to that we don't have error logs.

fsismondi commented:
Hello @hbollon thanks for the PR!  No idea why that was defined, I would drop it and see what happens

hbollon commented:
I used `camptocamp/argocd` in order to keep compatibility with my apps until I converted them to Helmfile because I've updated argocd in place on my cluster. However this custom image shouldn't be needed anymore.

fsismondi commented:
@hbollon Ill try to test this in my aks dev platform and be back with feedback

fsismondi commented:
for future reference `config.rbacConfig` chart value moved to `config.rbac`

fsismondi commented:
about rbac, delete the following:

```
        secret = {
         (...)
      -- rbacConfig = {
      --   scopes           = "[groups, cognito:groups, roles]"
      --   "policy.default" = ""
      --   "policy.csv"     = <<-EOT
      --                     g, pipeline, role:admin
      --                     g, argocd-admin, role:admin
      --                     EOT
      -- }
```

add instead:
```
      configs = {
        rbac = {
          scopes           = "[groups, cognito:groups, roles]"
          "policy.default" = ""
          "policy.csv"     = <<-EOT
                              g, pipeline, role:admin
                              g, argocd-admin, role:admin
                              EOT
        }
```

fsismondi commented:
dropping the ` | true`  of `helm dependency build | true` creates this situation:

```
module.prometheus-stack[0].module.kube-prometheus-stack.argocd_application.this: Creating...
module.prometheus-stack[0].module.kube-prometheus-stack.argocd_application.this: Still creating... [10s elapsed]
╷
│ Error: application kube-prometheus-stack could not be created
│ 
│   with module.prometheus-stack[0].module.kube-prometheus-stack.argocd_application.this,
│   on .terraform/modules/prometheus-stack/main.tf line 71, in resource "argocd_application" "this":
│   71: resource "argocd_application" "this" {
│ 
│ rpc error: code = InvalidArgument desc = application spec for kube-prometheus-stack is invalid: InvalidSpecError: Unable to generate manifests in charts/kube-prometheus-stack: rpc error: code = Unknown desc =
│ `/bin/sh -c helm dependency build` failed exit status 1: Error: no repository definition for https://prometheus-community.github.io/helm-charts. Please add the missing repos via 'helm repo add'
╵
Releasing state lock. This may take a few moments...

```

which seems to be related to [this issue](https://github.com/argoproj/argo-cd/issues/3451)
I just read the whole thread and got a headache afterwards, I propose leaving it as it as unless someone has a better idea

modridi commented on modules/values.tmpl.yaml:
```diff
@@ -105,7 +105,7 @@ argo-cd:
           generate:
             command: ["/bin/sh", "-c"]
-            args: ["echo \"$HELM_VALUES\" | helm template . --name-template $ARGOCD_APP_NAME --namespace $ARGOCD_APP_NAMESPACE $HELM_ARGS -f - --include-crds > all.yaml && kustomize build"]
+            args: ["echo \"$ARGOCD_ENV_HELM_VALUES\" | helm template . --name-template $ARGOCD_APP_NAME --namespace $ARGOCD_APP_NAMESPACE $HELM_ARGS -f - --include-crds > all.yaml && kustomize build"]
```
`HELM_ARGS` should also be prefixed.

hbollon commented:
Should be good now, I also moved `argo-cd.server.config` to `argo-cd.configs.cm` since it seems to have changed with the v5.7.0.
